digester.go

```diff
@@ -141,9 +141,12 @@ func newTarFilterer(writeCloser io.WriteCloser, filter func(hdr *tar.Header) (sk
 			closed = filterer.closed
 			filterer.closedLock.Unlock()
 		}
-		pipeReader.Close()
-		tarWriter.Close()
-		writeCloser.Close()
+		err1 := tarWriter.Close()
+		err := writeCloser.Close()
+		if err == nil {
+			err = err1
+		}
+		pipeReader.CloseWithError(err)
```

Comment:
I always worry excessively that a failing goroutine will leave the consumer hanging.

Consider something like
```go
func someOperationGoroutine() {
	err := errors.New("Internal error: unexpected panic in someOperationGoroutine")
	defer func() { // Ensure we _always_ wake up the caller
		_ = pipeReader.CloseWithError(err) // CloseWithError(nil) is equivalent to Close(), always returns nil
                wg.Done()
	}()
	err = someOperation()
}
func someOperation() {
    // straight-line worry-free Go code with liberally sprinkled early `return`s and any other control flow
}
```